test/BackendGeb/FromCore/Base.hs

```diff
@@ -34,55 +36,56 @@ coreToGebTranslationAssertion' coreInfoTable entryPoint expectedFile step = do
-              { _typedMorphism = translatedMorphism,
-                _typedMorphismObject = translatedObj
-              }
-      case run . runError @Geb.CheckingError $ Geb.check' typeMorph of
-        Left err ->
-          assertFailure . show . pretty $
-            fromJuvixError @GenericError (JuvixError err)
-        Right _ -> do
-          step "Try evaluating the JuvixCore node"
-          let resultCoreEval :: Core.Node = Core.evalInfoTable stderr readyCoreInfoTable
-          step "Translate the result of the evaluated JuvixCore node to Geb"
-          let (gebCoreEvalResult, _) = Geb.fromCore $ Core.setupMainFunction readyCoreInfoTable resultCoreEval
-          case ( Geb.eval' Geb.defaultEvalEnv translatedMorphism,
-                 Geb.eval' Geb.defaultEvalEnv gebCoreEvalResult
-               ) of
-            (Left err, _) -> do
-              step "The evaluation of the translated Geb node failed"
-              assertFailure . show . pretty $
-                fromJuvixError @GenericError (JuvixError err)
-            (_, Left err) -> do
-              step "The evaluation of gebCoreEvalResult failed"
-              assertFailure . show . pretty $ fromJuvixError @GenericError (JuvixError err)
-            ( Right resEvalTranslatedMorph,
-              Right resEvalGebCoreEvalResult
-              ) -> do
-                step "Compare the geb value of the Core eval output and the Geb eval output"
-                if
-                    | resEvalTranslatedMorph /= resEvalGebCoreEvalResult ->
-                        assertFailure "The evaluation for the Core node and the Geb node are not equal"
-                    | otherwise -> do
-                        let fpath = toFilePath expectedFile
-                        expectedInput <- TIO.readFile fpath
-                        step "Compare expected and actual program output"
-                        let compareEvalOutput morph =
-                              if
-                                  | Geb.quote resEvalTranslatedMorph /= morph ->
-                                      assertFailure $
-                                        "The result of evaluating the translated Geb"
-                                          <> "node is not equal to the expected output"
-                                  | otherwise -> assertBool "" True
-                        case Geb.runParser expectedFile expectedInput of
-                          Left parseErr -> assertFailure . show . pretty $ parseErr
-                          Right (Geb.ExpressionMorphism m) -> compareEvalOutput m
-                          Right (Geb.ExpressionTypedMorphism m) -> compareEvalOutput (m ^. Geb.typedMorphism)
-                          Right (Geb.ExpressionObject _) ->
-                            assertFailure "Expected a morphism, but got an object for the expected output"
+    Right readyCoreInfoTable ->
+      length (fromText (Core.ppTrace readyCoreInfoTable) :: String) `seq` do
```

Comment:
Why do you want to keep this here?

Reply:
To force full evaluation of the `toGeb` translation at this point. Otherwise, some parts of the result may get evaluated later, or even not at all if neither the type-checker nor the evaluator looks at them.